### PR TITLE
Demo-Application with IONIC

### LINKED ISSUE #2: Demo-App
Philipp is creating a Demo-Application with IONIC

### PR DESCRIPTION
**Pull request by philschmid**
# What does this PR do? 

This PR creates a initial APP set-up with `react-ionic` and `amplify`. I created the `amplify` directory top-level next to `frontend` I added some documentation for how to use `amplify` and pull the annoying, important `aws-exports.js` to run the application. I also started/create a frontend-specific documentation with tech stack and all relevant documentation sites. 

close #2

### CODE REVIEW

**Comment by philschmid** [comment]
Test Github Slack integration

**Comment by manzanillo** [comment]
package.json references react@17.0.1, while ionic is using @16.8.6
Shouldn't we also use react 16.8.6 instead of force use of react@17.0.0.1?

```
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: petrakip-app@0.0.1
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.6" from @ionic/react@5.5.0
npm ERR! node_modules/@ionic/react
npm ERR!   @ionic/react@"5.5.0" from the root project
```

**Comment by philschmid** [comment]
Did you use `npm install` or `yarn`? And I created the project with `ionic start myApp tabs --type react`

**Comment by manzanillo** [comment]
npm (npm throws errors), yarn warns about this. But if the ionic starter created the package json, I suggest just stick with it.

File: frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    "@testing-library/jest-dom": "5.11.9",
+    "@testing-library/react": "11.2.5",
+    "@testing-library/user-event": "12.6.3",
+    "@types/jest": "26.0.20",
+    "@types/node": "12.19.15",
+    "@types/react": "16.14.3",
+    "@types/react-dom": "16.9.10",
+    "@types/react-router": "5.1.11",
+    "@types/react-router-dom": "5.1.7",
+    "aws-amplify": "^3.3.27",
```
Comment (maxhaensel) [inline frontend/package.json:31]: probably devDependencies

File: frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    "ionicons": "5.4.0",
+    "react": "17.0.1",
+    "react-dom": "17.0.1",
+    "react-router": "5.2.0",
+    "react-router-dom": "5.2.0",
+    "react-scripts": "4.0.2",
+    "typescript": "4.1.3",
```
Comment (maxhaensel) [inline frontend/package.json:38]: probably devDependencies

File: frontend/src/components/header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { IonHeader, IonPage, IonTitle, IonToolbar } from '@ionic/react'
+import { PropsWithChildren } from 'react'
+
+interface HeaderProps {
+  pageTitle: string
+}
+
+const Header: React.FC<HeaderProps> = ({ pageTitle }) => {
```
Comment (maxhaensel) [inline frontend/src/components/header/index.tsx]: I would recommend to stick with ```Props``` and only use named interfaces when exporting these.

```suggestion
interface Props {
  pageTitle: string
}

const Header: React.FC<Props> = ({ pageTitle }) => {
```

**Review by maxhaensel — APPROVED** [review]

File: frontend/package.json (hunk 1 above)
Comment (philschmid) [reply frontend/package.json:31]: as mentioned ^ everything was created by `ionic-cli` and since we are building it as an app this doesn't matter.

**Pull request merged**